mintpy/tsview.py

```diff
@@ -711,7 +711,7 @@ def __init__(self, cmd=None, iargs=None):
         if cmd:
             iargs = cmd.split()[1:]
         self.cmd = cmd
-        self.iargs = iargs
+        self.iargs = cmd_line_parse(iargs)
         # print command line
         cmd = '{} '.format(os.path.basename(__file__))
         cmd += ' '.join(iargs)
```

Comment:
moving parsing to `__init__` (which I think also makes more sense) so that help and usage can be printed (and exit) gracefully if `iargs` is `None`

Reply:
`cmd_line_parse()` returns a Namespace object, while `self.iargs` is a list of strings, so I don't think this modification works.

Reply:
If `iargs` is `None`, because [`main` got called with `iargs=None` ](https://github.com/jhkennedy/MintPy/blob/entrypints/mintpy/tsview.py#L1025-L1026), then without this change, `timeseriesViewer.__init__()` will crash out when [trying to form the `cmd` string](https://github.com/jhkennedy/MintPy/blob/entrypints/mintpy/tsview.py#L717) becuase:
```python
>>> iargs = None
>>> ' '.join(iargs)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only join an iterable
>>> 
```
So either you need a:
```python
if iargs is not None:
    cmd += ' '.join(iargs)
```

or to handle the parsing before this point. The benefit of parsing as early as possible, is that you'll exit *fast* (before other errors could happen) for `--help`/missing/invalid arguments. 

it doesn't look like `self.iargs` is used anywhere except in the later configure step, and so the change below accommodates the change to a `Namespace` object. If it's used/access elsewhere, the `if` block  might be more appropriate.

Reply:
Thanks for the explanation. Your `if` block looks good to me. I saved `self.iargs` specifically to check input options later on in other scripts/notebooks (should have said this earlier; view.py has the same setup as well).

Reply:
Switched to the `if` block!